packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx

```diff
@@ -55,6 +63,30 @@ export const BookEventForm = ({ onCancel }: BookEventFormProps) => {
   const event = useEvent();
   const eventType = event.data;
 
+  const reserveSlot = () => {
+    if (eventType) {
+      reserveSlotMutation.mutate({
+        slotUtcStartDate: dayjs(timeslot).utc().format(),
+        eventTypeId: eventType.id,
+        slotUtcEndDate: dayjs(timeslot)
+          .utc()
+          .add(duration || eventType.length, "minutes")
+          .format(),
+      });
+    }
+  };
+  useEffect(() => {
+    reserveSlot();
+    const interval = setInterval(reserveSlot, parseInt(MINUTES_TO_BOOK) * 60 * 1000 - 2000);
+    return () => {
+      if (eventType) {
+        releaseSlotMutation.mutate();
+        clearInterval(interval);
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [eventType]);
```

Comment:
reserve the slot on the booking form mount, and release on unmount the component